### PR TITLE
ARROW-2705: [JS] CombinationPredicates should take list of predicates

### PR DESCRIPTION
**Pull request by TheNeuralBit**
`CombinationPredicate` now stores a list of child predicates rather than just a left and right
Also adds `and`, `or` functions for easier creation of combination predicates with
several children

### CODE REVIEW

**Comment by wesm** [comment]
@trxcllnt what do you think?

File: js/src/predicate.ts [hunk 1]
```diff
@@ -77,10 +77,9 @@ export class Col<T= any> extends Value<T> {
 
 export abstract class Predicate {
     abstract bind(batch: RecordBatch): PredicateFunc;
-    and(expr: Predicate): Predicate { return new And(this, expr); }
-    or(expr: Predicate): Predicate { return new Or(this, expr); }
+    and(...expr: Predicate[]): And { return new And(this, ...expr); }
+    or(...expr: Predicate[]): And { return new Or(this, ...expr); }
```
Comment (trxcllnt) [inline js/src/predicate.ts]: Should the return type here be `: Or`?

File: js/src/predicate.ts [hunk 2]
```diff
@@ -256,6 +270,8 @@ export class CustomPredicate extends Predicate {
 
 export function lit(v: any): Value<any> { return new Literal(v); }
 export function col(n: string): Col<any> { return new Col(n); }
+export function and(...p: Predicate[]): Predicate { return new And(...p); }
+export function or(...p: Predicate[]): Predicate { return new Or(...p); }
```
Comment (trxcllnt) [inline js/src/predicate.ts]: Should these return types also be `And` and `Or` respectively, to match the definitions above?

**Review by trxcllnt — COMMENTED** [review]
a few questions about types, but other than that looks great

File: js/src/predicate.ts (hunk 1 above)
Comment (TheNeuralBit) [reply js/src/predicate.ts]: wow, not sure how I got away with that mistake

**Comment by TheNeuralBit** [comment]
@trxcllnt fixed the type issues, I think this is good to merge once CI finishes

**Comment by TheNeuralBit** [comment]
CI failure sems unrelated, I'll go ahead and merge

**Pull request closed without merging**